include/common/bitdepth.rs

```diff
@@ -96,6 +97,16 @@ impl BPC {
             Self::BPC16
         }
     }
+
+    pub fn pxstride<T>(self, n: T) -> T
+    where
+        T: Copy + From<u8> + Div<Output = T>,
+    {
+        match self {
+            BPC::BPC8 => n,
+            BPC::BPC16 => n / 2.into(),
+        }
+    }
```

Comment:
This is good but I want to slightly rework things so we aren't redundant.  I'll open a PR.

Reply:
See #780.